pandas/core/apply.py

```diff
@@ -44,7 +49,38 @@ def frame_apply(
 
 
 class FrameApply:
-    def __init__(self, obj, func, raw, result_type, ignore_failures, args, kwds):
+
+    # ---------------------------------------------------------------
+    # Abstract Methods
+
+    @property
+    def axis(self) -> int:
+        raise AbstractMethodError(self)
+
+    @property
+    def result_index(self) -> "Index":
+        raise AbstractMethodError(self)
+
+    @property
+    def result_columns(self) -> "Index":
+        raise AbstractMethodError(self)
+
+    @property
+    def series_generator(self) -> Iterator["Series"]:
```

Comment:
For return types, this could also be typed as `Iterable`. we've done this in a couple of places.

Reply:
isnt iterator more informative than iterable?

Reply:
for a return type it's nor relevant, so it's just a consistency thing.